mobile/src/reducers/auth_reducer.js

```diff
@@ -21,6 +22,15 @@ export default function (state = initialState, action) {
       return { ...state, counterLogoutTime: action.payload, resetCounter: !state.resetCounter };
     case AUTH_ERROR:
       return { ...state, error: action.payload, authenticated: false };
+    case STORE_LOGIN: {
+      const { username, hash, token } = action.payload;
+      const newState = Object.assign({}, state);
```

Comment:
I think this could work here: 
{ ...state, savedLogins: { ...state.savedLogins, savedLogins[username]: { hash, token } }